presto-main/src/main/java/com/facebook/presto/transaction/TransactionManager.java

```diff
@@ -279,6 +279,13 @@ public void fail(TransactionId transactionId)
     @ThreadSafe
     private static class TransactionMetadata
     {
+        private enum State
```

Comment:
so im not convinced i like this structure necessarily. I chose the pre-existing atomic ref to boolean to show that the state can only move from null to a single terminal state (of which the code enforces only 2 possible terminate states). If this is for readability, why not just change the AtomicRef of a boolean to a state with 2 terminal values?